packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx

```diff
@@ -1,20 +1,73 @@
 import * as React from 'react';
 import {Box} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
+import type {BoxStyleProps} from '@twilio-paste/box';
 import {CloseIcon} from '@twilio-paste/icons/esm/CloseIcon';
 
+type PillVariant = 'error' | 'default';
+
 interface PillCloseIconProps {
-  onClick: () => void;
+  onClick?: () => void;
+  selected?: boolean;
+  variant?: PillVariant;
+  element?: BoxProps['element'];
 }
 
-export const PillCloseIcon: React.FC<PillCloseIconProps> = ({onClick}) => {
+const hoverStyles: Record<PillVariant, BoxStyleProps> = {
+  default: {
+    backgroundColor: 'transparent',
+    borderColor: 'colorBorderPrimaryStronger',
+  },
+  error: {
+    backgroundColor: 'transparent',
+    borderColor: 'colorBorderErrorStronger',
+  },
+};
+
+const selectedHoverStyles: Record<PillVariant, BoxStyleProps> = {
+  default: {
+    borderColor: 'transparent',
+    backgroundColor: 'colorBackgroundPrimary',
+  },
+  error: {
+    borderColor: 'transparent',
+    backgroundColor: 'colorBackgroundErrorStronger',
+  },
+};
+
+export const PillCloseIcon: React.FC<PillCloseIconProps> = ({
+  element = 'FORM_PILL_CLOSE',
+  onClick = () => {},
+  selected = false,
+  variant = 'default',
+}) => {
+  const computedHoverStyles = selected ? selectedHoverStyles[variant] : hoverStyles[variant];
+
   return (
     <Box
+      element={element}
       onClick={onClick}
+      position="absolute"
+      right={0}
+      top={0}
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+      height="24px"
+      width="24px"
```

Comment:
question: So for here, given this is an icon, though not a fully blown paste icon, why wouldn't we be using a icon size e.g. `$size-icon-40` (24px).

Then using that as our reference for a height we need to align to?

Reply:
Oh, that works! Will use the existing token.